### PR TITLE
Bug fixes for wi24

### PR DESCRIPTION
**Pull request by JustKong13**
### Summary 

This pull request fixes a few bugs and prepares LSC for the winter semester, as well as 
- [x] Dashboard no longer preloads spring 23
- [x] Fixed Email Regex to have to end with `cornell.edu`
- [x] Added winter '24 semester to metrics 

### Test Plan 

Metrics page also has winter '24 as a valid option
Screenshot 2024-01-03 at 5 16 22 PM

Now correctly identifies invalid emails
Screenshot 2024-01-03 at 5 15 37 PM

### Notes 

In the future we will need to make the metrics semesters update automatically. 

### Breaking Changes 

`yarn.lock` was updated when I did `yarn install`, it seems to me that the changes primarily consists of a new syntax update which just removed quotation marks. 
`package-lock.json` was also updated when I did `npm install`, due to me working off of a new machine. There were no deletions or additions, just standard upgrades.

### CODE REVIEW

File: backend/functions/src/student/functions.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export const addStudentSurveyResponse = async (
   }
 
   // 0. Check if email is valid cornell.edu email.
-  const emailRegex = /^\w+@cornell.edu$/
+  const emailRegex = /^\w+@cornell\.edu$/
```
Comment (rgu0114) [inline backend/functions/src/student/functions.ts:115]: idr how regex works oop, does this like, make sure emails end with explicitly .edu and not like .education?

**Review by rgu0114 — COMMENTED** [review]
Maybe mention what dependency changes u made in package.lock/yarn.lock

File: backend/functions/src/student/functions.ts (hunk 1 above)
Comment (Coreball) [reply backend/functions/src/student/functions.ts:115]: Probably has to do with the dot matching any character other than line terminators lol

Comment (rgu0114) [reply backend/functions/src/student/functions.ts:115]: ah ok. btw, hi Changyuan!

**Comment by JustKong13** [comment]
> Maybe mention what dependency changes u made in package.lock/yarn.lock

**Comment by JustKong13** [comment]
> Maybe mention what dependency changes u made in package.lock/yarn.lock

Just added that to the pull request! Basically I think for frontend there was an update which removed quotation marks around certain parts of the file (with no meaningful dependency changes), and for the backend our packages had version upgrades. There was no addition or deletion of any packages

**Review by rgu0114 — APPROVED** [review]
looks good 👍

**Pull request merged**